src/plugins/safeSnap/utils/events.ts

```diff
@@ -129,14 +130,20 @@ export async function pageEvents<E>(
   endBlock: number,
   maxRange: number,
   //start and end block range to query
-  fetchEvents: (query: { start: number; end: number }) => Promise<E[]>
+  fetchEvents: (query: { start: number; end: number }) => Promise<E[]>,
+  concurrency: number = 5
 ): Promise<E[]> {
   let state = rangeStart({ startBlock, endBlock, maxRange });
   const ranges: { start: number; end: number }[] = [];
+  let index = 0;
   do {
-    ranges.push({ start: state.currentStart, end: state.currentEnd });
+    ranges.push({
+      start: state.currentStart,
+      end: state.currentEnd,
+      index: index++
+    });
     state = rangeSuccessDescending(state);
   } while (!state.done);
 
-  return (await Promise.all(ranges.map(fetchEvents))).flat();
+  return (await bb.map(ranges, fetchEvents, { concurrency })).flat();
```

Comment:
this is really the main change, this limits promises being executed to `concurrency` amount, by default 5.  performance is still quite good in terms of total loading time.